src/NexusMods.Networking.GOG/Client.cs

```diff
@@ -241,6 +242,106 @@ public bool TryGetAuthInfo(out AuthInfo.ReadOnly authInfo)
+            }).ToArray();
+
+            return result;
+        }, cancellationToken: token);
+    }
+
+    public async ValueTask DownloadInstallerArchive((OSPlatform OS, Uri DownloadLink) installerInfo, Stream output, CancellationToken token)
+    {
+        if (installerInfo.OS != OSPlatform.Linux) throw new NotSupportedException();
+        // NOTE(erri120): Linux installers are downloaded as shell scripts.
+        // The shell script is a self-extractable archive created using makeself (https://makeself.io/)
+        // A startup script is configured to run a mojosetup script (https://icculus.org/mojosetup/)
+        // We don't really care about any of this since we can just get the game contents directly by finding
+        // the ZIP archive embedded in the downloaded file and only extracting the game files.
+
+        await _pipeline.ExecuteAsync(async cancellationToken =>
+        {
+            Uri downloadLink;
+
+            using (var msg = await CreateMessage(installerInfo.DownloadLink, cancellationToken))
+            using (var response = await _client.SendAsync(msg, cancellationToken: cancellationToken))
+            {
+                var installerResponse = await response.Content.ReadFromJsonAsync<InstallerResponse>(_jsonSerializerOptions, cancellationToken: cancellationToken);
+                if (installerResponse is null) throw new JsonException("Failed to get installer");
+                downloadLink = new Uri(installerResponse.DownloadLink);
+            }
+
+            using (var msg = await CreateMessage(downloadLink, cancellationToken))
+            using (var response = await _client.SendAsync(msg, cancellationToken: cancellationToken))
+            {
+                await using var stream = await response.Content.ReadAsStreamAsync(cancellationToken: cancellationToken);
+                await CopyArchive(stream, output, cancellationToken);
+            }
+        }, cancellationToken: token);
+    }
+
+    private static async ValueTask CopyArchive(Stream input, Stream output, CancellationToken cancellationToken)
+    {
+        var bufferSize = Size.MB;
+        var buffer = new Memory<byte>(array: new byte[bufferSize.Value]);
+
+        var found = false;
+
+        while (true)
+        {
+            var numRead = await input.ReadAsync(buffer, cancellationToken: cancellationToken);
+            if (numRead <= 0) break;
+
+            var span = buffer.Span.Slice(start: 0, length: numRead);
+            if (!IsArchive(span, out var index)) continue;
+
+            found = true;
+            await output.WriteAsync(buffer[index..], cancellationToken: cancellationToken);
+            break;
+        }
+
+        if (!found) throw new NotSupportedException("The input Stream doesn't appear to contain an archive");
+        await input.CopyToAsync(output, cancellationToken: cancellationToken);
+    }
+
+    private static bool IsArchive(ReadOnlySpan<byte> span, out int index)
```

Comment:
This is verifying a zip magic header.
Can you just make a quick comment noting that; it may not be immediately obvious.